pubsublite-beam-io/src/main/java/com/google/cloud/pubsublite/beam/SubscriberOptions.java

```diff
@@ -133,24 +143,54 @@ public abstract Builder setSubscriberStubSupplier(
     public abstract Builder setCommitterStubSupplier(
         SerializableSupplier<CursorServiceStub> stubSupplier);
 
+    public abstract Builder setTopicBacklogReader(TopicBacklogReader topicBacklogReader);
+
+    // Used in unit tests
     abstract Builder setSubscriberFactory(SubscriberFactory subscriberFactory);
 
     abstract Builder setCommitterSupplier(SerializableSupplier<Committer> committerSupplier);
 
+    // Used for implementing build();
```

Comment:
Does this work? Cool trick if it does.

Reply:
Yeah, it's pretty cool, I found it here[1]! In one iteration of this I had tests and it was working then

[1] https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#-normalize-modify-a-property-value-at-build-time